js/src/evaluation/evaluator.ts

```diff
@@ -225,3 +240,45 @@ export class DynamicRunEvaluator<Func extends (...args: any[]) => any>
 export function runEvaluator(func: RunEvaluatorLike): RunEvaluator {
   return new DynamicRunEvaluator(func);
 }
+
+// Define the object parameter type
+type EvaluatorObjectParams = {
+  run?: Run;
+  example?: Example;
+  inputs?: Record<string, any>;
+  outputs?: Record<string, any>;
+  referenceOutputs?: Record<string, any>;
+};
+
+// Define all possible evaluator function signatures
+type EvaluatorFunction =
+  | ((run: Run, example?: Example) => EvaluationResult | EvaluationResults)
+  | ((
+      run: Run,
+      example?: Example
+    ) => Promise<EvaluationResult | EvaluationResults>)
+  | ((params: EvaluatorObjectParams) => EvaluationResult | EvaluationResults)
+  | ((
+      params: EvaluatorObjectParams
+    ) => Promise<EvaluationResult | EvaluationResults>);
+
+function getEvaluatorParameters(func: EvaluatorFunction): {
+  type: "tuple" | "object";
+} {
+  const funcStr = func.toString();
+
+  // Check if the function accepts a single object parameter
+  if (funcStr.match(/^\s*(?:async\s+)?(?:function\s*)?\(?\s*{\s*[a-zA-Z_]/)) {
+    return { type: "object" };
+  }
+
+  // Check if the function accepts run and example parameters
+  if (
+    funcStr.match(/^\s*(?:async\s+)?(?:function\s*)?\(?\s*run\s*,\s*example/)
```

Comment:
You definitely don't want to do this in general - will break if someone does something as simple as renames parameters

Reply:
i think thats fine, in py we expect specific params

Reply:
Positional args aren't named in JS